### PR TITLE
Enable colored compiler diagnostics with the background process as well

### PR DESCRIPTION
**Pull request by jimhester**
Fixes https://github.com/r-lib/pkgdepends/issues/138

### CODE REVIEW

File: R/build.R [hunk 1]
```diff
@@ -87,6 +87,7 @@ build_setup <- function(path, dest_path, binary, vignettes, manual, clean_doc, a
 
   if (needs_compilation) {
     update_registration(path, compile_attributes, register_routines, quiet)
+    withr:::local_makevars(compiler_flags(debug = FALSE), assignment = "+=", .local_envir = env)
```
Comment (jimhester) [inline R/build.R:90]: `local_makevars` was I think inadvertently not exported, I need to do a withr release soon anyway, so we can export it then.

**Comment by jimhester** [comment]
This adds the compiler flags for both the interactive builds and the background process. When the interactive builds use the `parent.frame()` as the cleanup environment, the background uses the R6 object, which I think will do what we want.

**Review by gaborcsardi — COMMENTED** [review]
Cool! I think the CI errors might be related, no?

**Comment by jimhester** [comment]
Yeah, likely due to when the cache gets invalidated, I will look into it.

**Pull request merged**